src/solvers/Solver.py

```diff
@@ -71,24 +72,21 @@ def get_parameter_options(self) -> dict:
         :rtype: dict
         """
         pass
-        
+
     def get_submodule(self, device_option: str) -> any:
         """
-        If self.sub_options is not None a device is instantiated according to the information given in self.sub_options.
-        Otherwise get_device is called as fall back.
-        
-        :param mapping_option: String with the option
-        :rtype: str
-        :type mapping_option: str
+        If self.sub_options is not None, a device is instantiated according to the information given in self.sub_options.
+        Otherwise, get_device is called as fall back.
+
+        :param device_option: String with the option
+        :type device_option: str
         :return: instance of the device class
         :rtype: any
         """
-        
         if self.sub_options is None:
             return self.get_device(device_option)
         else:
-            return _get_instance_with_sub_options(self.sub_options, device_option, device_option)
-        
+            return _get_instance_with_sub_options(self.sub_options, device_option)
```

Comment:
```suggestion
            return _get_instance_with_sub_options(self.sub_options, device_option, device_option)
```

Let`s keep it as it was for now, thank for the heads up @jusschwitalla ! 

After this is merged I will push a PR which allows the user to specify a different device_name in the modules file.